enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,641 @@
+* Expose the Assisted Installer's capabilities as a kubernetes-native API.
+* Enable any OpenShift cluster deployed by the Assisted Installer to add worker nodes on day 2.
+* Enable multi-cluster management tooling to create new clusters using Assisted Installer.
+* Utilize [metal3](https://metal3.io/) to boot the assisted discovery ISO for bare-metal deployments.
+* Ensure the design is be extensible for installing on other platforms.
+* Utilize the Assisted Service and Agent with minimal changes to avoid
+adversely affecting the cloud.redhat.com deployment.  While making a version
+that is Kubernetes-native with no SQL DB is on the table for the long term,
+it is not feasible to implement and maintain two versions in the near-term.
+
+### Non-Goals
+
+* Solve central machine management. That can be done with this effort, or after
+this effort, but it is not strictly a requirement in order to deliver the
+goals stated above.
+* Run metal3 components on a non-baremetal cluster.
+
+## Proposal
+
+### User Stories
+
+#### Day 2 Add Worker
+
+After deploying a cluster with Assisted Installer, I can add workers from bare
+metal hardware on day two using a similar workflow and tool set. I can either
+obtain the discovery live ISO and use my own methods to boot it, or I can use
+the baremetal-operator to boot the live ISO automatically.
+
+#### Multi-cluster
+
+As a user of Red Hat's multi-cluster mangement tools, I can use the assisted
+installation agent-based workflow to create clusters from a pool of bare metal
+inventory.
+
+### Implementation Details/Notes/Constraints
+
+#### Assisted Installer k8s-native API
+
+The Assisted Installer has a [REST
+API](https://generator.swagger.io/?url=https://raw.githubusercontent.com/openshift/assisted-service/master/swagger.yaml)
+that is available at cloud.redhat.com. The service is implemented with a
+traditional relational database. In order to integrate with OpenShift
+infrastructure management tooling, it is necessary to additionally expose its
+capabilities as Kubernetes-native APIs. They will be implemented as Custom
+Resource Definitions and a controller with the assisted service's capabilities.
+
+For the first deliverable, a local deployment of the backing service will
+include new controllers that will operate to achieve the desired state as
+declared in CRDs. Thus the service will expose both REST and Kubernetes-native
+APIs, and both API layers will call the same set of internal methods. The local
+service will include a database; sqlite if possible, or else postgresql.
+
+    --------------------------
+    |  REST API  |  k8s API  |
+    --------------------------
+    | Assisted Service logic |
+    --------------------------
+        |               |
+        v               v
+      SQL DB        File System
+                      storage
+
+For modest clusters that do not have persistent storage available but need
+day-2 ability to add nodes, the database will need to be reconstructable in
```

Comment:
The idea was to require persistent storage for the multi-cluster case which is more complex, and there is already a precedent for (ACM).  However, it would be a difficult to tell users that they need persistent storage to add workers.